src/utils/proxy.js

```diff
@@ -378,15 +367,36 @@ const startProxy = async function (settings, addonsUrls, configPath, projectDir)
     }
 
     proxy.web(req, res, options)
+  }
+
+  return onRequest
+}
+
+const createSecureServer = ({ onRequest, cert, key }) => https.createServer({ cert, key }, onRequest)
+
+const startProxy = async function (settings, addonsUrls, configPath, projectDir) {
+  const functionsServer = settings.functionsPort ? `http://localhost:${settings.functionsPort}` : null
+
+  const proxy = initializeProxy(settings.frameworkPort, settings.dist, projectDir)
+
+  const rewriter = await createRewriter({
+    distDir: settings.dist,
+    jwtSecret: settings.jwtSecret,
+    jwtRoleClaim: settings.jwtRolePath,
+    configPath,
+    projectDir,
   })
 
+  const onRequest = getRequestListener({ proxy, rewriter, settings, addonsUrls, functionsServer })
+  const server = settings.https ? createSecureServer({ ...settings.https, onRequest }) : http.createServer(onRequest)
   server.on('upgrade', function onUpgrade(req, socket, head) {
     proxy.ws(req, socket, head)
   })
 
   return new Promise((resolve) => {
     server.listen({ port: settings.port }, () => {
```

Comment:
I am not sure whether this is something to fix separately, but we should promisify this with `await promisify(server.listen.bind(server))({ port: settings. port })`. This would also allow handle errors (e.g. port errors).

Reply:
Done in https://github.com/netlify/cli/pull/2132/commits/ec5c40a204502cb11960aa7ee15387ce60f6cc18

As discussed we can't use `promisify` as errors are emitted via `server.on('error')`